includes/class-wc-stripe-intent-controller.php

```diff
@@ -266,10 +266,7 @@ public function create_payment_intent_ajax() {
 		try {
 			$is_nonce_valid = check_ajax_referer( '_wc_stripe_nonce', false, false );
 			if ( ! $is_nonce_valid ) {
-				throw new Process_Payment_Exception(
-					__( "We're not able to process this payment. Please refresh the page and try again.", 'woocommerce-gateway-stripe' ),
-					'wc_stripe_upe_intent_error'
-				);
+				throw new Exception( __( "We're not able to process this payment. Please refresh the page and try again.", 'woocommerce-gateway-stripe' ) );
```

Comment:
Is there a reason to revert to a generic `Exception`? I think in general it would be beneficial to keep the more specific `Process_Payment_Exception` here to give more context 🤔

Reply:
The class `Process_Payment_Exception` does not exist in this repo, was a copy/paste mistake when copying the UPE code from the WC Payments repo (in the `feature/upe` PR)

In the plugin, we have only `WC_Stripe_Exception` but is just a wrapper with two messages (original and translated).
We shpuld do like in WC Payments, create a base exception class with a `message` and a string `error_code` and then create specific exceptions, in this case, `Process_Payment_Exception` with code `wc_stripe_upe_intent_error`.

But this PR is getting too big already, maybe we can add them in another PR?

Reply:
Ahhh, sorry about that! I thought you were removing something that we had in Stripe already!

> this PR is getting too big already, maybe we can add them in another PR?

Yes, I agree. Let's just open a new issue for this since this is a more generic "use custom exceptions" issue than something specific to UPE or this PR 👍